src/native/corehost/hostmisc/utils.cpp

```diff
@@ -350,6 +350,56 @@ bool try_stou(const pal::string_t& str, unsigned* num)
     return true;
 }
 
+#if defined(_WIN32)
+bool is_running_in_wow64()
+{
+    BOOL fWow64Process = FALSE;
+    if (!IsWow64Process(GetCurrentProcess(), &fWow64Process))
+    {
+        return false;
+    }
+    return (fWow64Process != FALSE);
+}
+
+typedef BOOL (WINAPI* is_wow64_process2)(
+  HANDLE hProcess,
+  USHORT *pProcessMachine,
+  USHORT *pNativeMachine
+);
+
+bool is_emulating_x64()
+{
+    USHORT pProcessMachine, pNativeMachine;
+    auto kernel32 = LoadLibraryEx("kernel32.dll", NULL, 0);
+    if (kernel32)
+    {
+        is_wow64_process2 isWow64Process2Func = (is_wow64_process2)GetProcAddress(kernel32, "IsWow64Process2");
+        if (!isWow64Process2Func)
+        {
+            // Could not find IsWow64Process2.
+            FreeLibrary(kernel32);
```

Comment:
Technically this should free the library after every call.

It might be a good idea to actually cache the result of this whole function in a static variable. The result should never change. But it's probably not as important (we would need to make it thread safe BTW).

Reply:
You do not need to bother with freeing kernel32.dll. It is not an unloadable library.